postgres/src/test/scala/zio/sql/postgresql/FunctionDefSpec.scala

```diff
@@ -1090,6 +1090,34 @@ object FunctionDefSpec extends PostgresRunnableSpec with DbSchema {
 
       val result = typeCheck("execute((select(CharLength(Customers.fName))).to[Int, Int](identity))")
       assertZIO(dummyUsage *> result)(isLeft)
+    },
+    test("date-trunc woth hour") {
+      val someInstant = Instant.parse("1997-05-07T10:15:30.00Z")
+      val query       = select(DateTrunc("hour", someInstant))
+      val testResult  = execute(query)
+
+      val assertion =
+        for {
+          r <- testResult.runCollect
+        } yield assert(timestampFormatter.format(r.head))(
+          Assertion.matchesRegex("[0-9]{4}-[0-9]{2}-[0-9]{2} 10:00:00.0000")
```

Comment:
What is the reason for using the regex to match the date part here?

Reply:
Initially, I compared the actual dates/time. But it was offset by 1hr. 
I assumed it was Timezone difference between the Test-Container used for the test and my JDK. 
Noticed some other test in the codebase used regex so I went with it.

Reply:
But I see the time part is constant in this matching so probably the date one even more. I guess the regex is not needed here. 

> Noticed some other test in the codebase used regex so I went with it.

Thos could be about kinda random things like "now" or "current time". Here a test is matching the exact data and exact time, so a straightforward matching shall be sufficient.